Author: Nicholas Bamber <nicholas@periapt.co.uk>
Subject:
 - Perl script failing in Makefile rule due to lack of permissions
 - not building XML stuff in same source directory
Last-Update: 2012-10-12
Forwarded: no
--- a/Makefile
+++ b/Makefile
@@ -20,13 +20,10 @@
 # Override the above variables in config.mk if needed.
 -include config.mk
 
-all: htdocs/docs/errors.html htdocs/sgml-lib/catalog.xml $(GZIP_FILES)
+all: htdocs/docs/errors.html $(GZIP_FILES)
 
 htdocs/docs/errors.html: misc/docs_errors.pl share/templates/en_US/error_messages.cfg share/templates/en_US/docs_errors.tmpl htdocs/config/validator.conf
-	env W3C_VALIDATOR_HOME=. W3C_VALIDATOR_CFG=htdocs/config/validator.conf misc/docs_errors.pl > $@
-
-htdocs/sgml-lib/catalog.xml: misc/soc2xml.pl htdocs/sgml-lib/xml.soc
-	misc/soc2xml.pl < htdocs/sgml-lib/xml.soc > $@
+	env W3C_VALIDATOR_HOME=. W3C_VALIDATOR_CFG=htdocs/config/validator.conf perl -T misc/docs_errors.pl > $@
 
 htdocs/scripts/combined.js: $(JS_SOURCES)
 	rm -f $@
